Core/Src/main.c

```diff
@@ -87,14 +91,45 @@ int main(void)
   /* Initialize all configured peripherals */
   MX_GPIO_Init();
   MX_USART2_UART_Init();
+  MX_SPI1_Init();
+  MX_TIM1_Init();
   /* USER CODE BEGIN 2 */
 
+  // {ADC start, CH0, don't care}
+  uint8_t tx_data[COMMS_LENGTH] = {0b0000001, 0b1000000, 0b0000000};
```

Comment:
Correct tx data! And just a recommendation here, usually we don't write numbers in binary form, hex is more often used since it is more readable. I already saw you using it below, which is great.